skxray/core/speckle.py

```diff
@@ -0,0 +1,295 @@
+"""
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import six
+import numpy as np
+import time
+
+import skxray.core.correlation as corr
+import skxray.core.roi as roi
+from skxray.core.utils import bin_edges_to_centers, geometric_series
+
+import logging
+logger = logging.getLogger(__name__)
+
+
+def xsvs(image_sets, label_array, number_of_img, timebin_num=2,
+         max_cts=None):
+    """
+    This function will provide the probability density of detecting photons
+    for different integration times.
+
+    The experimental probability density P(K) of detecting photons K is
+    obtained by histogramming the speckle counts over an ensemble of
+    equivalent pixels and over a number of speckle patterns recorded
+    with the same integration time T under the same condition.
+
+    Parameters
+    ----------
+    image_sets : array
+        sets of images
+    label_array : array
+        labeled array; 0 is background.
+        Each ROI is represented by a distinct label (i.e., integer).
+    number_of_img : int
+        number of images (how far to go with integration times when finding
+        the time_bin, using skxray.utils.geometric function)
+    timebin_num : int, optional
+        integration time; default is 2
+    max_cts : int, optional
+       the brightest pixel in any ROI in any image in the image set.
+       defaults to using skxray.core.roi.roi_max_counts to determine the brightest
+       pixel in any of the ROIs
+
+    Returns
+    -------
+    prob_k_all : array
+        probability density of detecting photons
+    prob_k_std_dev : array
+        standard deviation of probability density of detecting photons
+
+    Notes
+    -----
+    These implementation is based on following references
+    References: text [1]_, text [2]_
+
+    .. [1] L. Li, P. Kwasniewski, D. Oris, L Wiegart, L. Cristofolini,
+       C. Carona and A. Fluerasu , "Photon statistics and speckle visibility
+       spectroscopy with partially coherent x-rays" J. Synchrotron Rad.,
+       vol 21, p 1288-1295, 2014.
+
+    .. [2] R. Bandyopadhyay, A. S. Gittings, S. S. Suh, P.K. Dixon and
+       D.J. Durian "Speckle-visibilty Spectroscopy: A tool to study
+       time-varying dynamics" Rev. Sci. Instrum. vol 76, p  093110, 2005.
+
+    There is an example in https://github.com/scikit-xray/scikit-xray-examples
+    It will demonstrate the use of these functions in this module for
+    experimental data.
+
+    """
+    if max_cts is None:
+        max_cts = roi.roi_max_counts(image_sets, label_array)
+
+    # find the label's and pixel indices for ROI's
+    labels, indices = corr.extract_label_indices(label_array)
+
+    # number of ROI's
+    u_labels = list(np.unique(labels))
+    num_roi = len(u_labels)
+
+    # create integration times
+    time_bin = geometric_series(timebin_num, number_of_img)
+
+    # number of items in the time bin
```

Comment:
I think either one is correct, if your thing time is more suitable I will change it